phpBB/phpbb/group/helper.php

```diff
@@ -37,4 +78,198 @@ public function get_name($group_name)
 	{
 		return $this->language->is_set('G_' . utf8_strtoupper($group_name)) ? $this->language->lang('G_' . utf8_strtoupper($group_name)) : $group_name;
 	}
+
+	/**
+	 * Get group name details for placing into templates.
+	 *
+	 * @param string	$mode				Can be profile (for getting an url to the profile), group_name (for obtaining the group name), colour (for obtaining the group colour), full (for obtaining a html string representing a coloured link to the group's profile) or no_profile (the same as full but forcing no profile link)
+	 * @param int		$group_id			The group id
+	 * @param string	$group_name			The group name
+	 * @param string	$group_colour		The group colour
+	 * @param mixed		$custom_profile_url	optional parameter to specify a profile url. The group id gets appended to this url as &amp;g={group_id}
+	 *
+	 * @return string A string consisting of what is wanted based on $mode.
+	 */
+	public function get_name_string($mode, $group_id, $group_name, $group_colour = '', $custom_profile_url = false)
+	{
+		$_profile_cache = array(
```

Comment:
I see this used to be a static variable. Instead of redeclaring this every every time you can just add it as a class member variable.

Reply:
Adjusted